### PR TITLE
Tried to fix issue #173 (using pcre2posix.h in MariaDB)

### PR DESCRIPTION
**Pull request by cbouc**
#173 Added default (empty) definition for PCRE2_CALL_CONVENTION in prec2posix.h just like  that in prec2.h

### CODE REVIEW

File: src/pcre2posix.h [hunk 1]
```diff
@@ -116,6 +116,20 @@ typedef struct {
...
+void __cdecl function(....)
+
+might be needed. In order so make this easy, all the exported functions have
+PCRE2_CALL_CONVENTION just before their names. It is rarely needed; if not
+set, we ensure here that it has no effect. */
+
+#ifndef PCRE2_CALL_CONVENTION
+#define PCRE2_CALL_CONVENTION
+#endif
+
```
Comment (carenas) [inline src/pcre2posix.h:132]: the equivalent code should be then removed from pcre2posix.c

**Review by carenas — CHANGES_REQUESTED** [review]

**Comment by PhilipHazel** [comment]
Unfortunately, this issue was not caught by the tests because pcre2test includes both pcre2.h and pcre2posix.h. I will do two things: I will move the default definition (and its comment) from pcre2posix.c to pcre2posix.h and I will write a little test program that just includes pcre2posix.h (without pcre2.h) and add this test to the test suite.

**Comment by thesamesam** [comment]
@PhilipHazel Once a test is added, would you consider a new release for this? Otherwise all distributors of pcre2 are likely to have to hit the problem and then backport the fix after hopefully stumbling upon the bug I filed.

**Comment by PhilipHazel** [comment]
Fixed by 16802f39. I will put out a new release fairly soon.

**Pull request closed without merging**